mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java

```diff
@@ -53,35 +54,65 @@
      *                  to directly use the system malloc/free functions.
      *      + default value: 0
      * 3. wait policy:
-     *      + function: Sets omp wait policy to passive.
+     *      + function: Sets omp wait policy to passive. passive or active
      *      + default value: passive
      * 4. disable fast mm:
      *      + function: Sets the time (milliseconds) that a thread should wait before sleeping,
      *                  after completing the execution of a parallel region.
      *      + default value: true
      */
     private static void setMklEnv() {
-        String mklBlockTimeStr = System.getProperty("bigdl.mklBlockTime", "0");
-        int mklBlockTime = Integer.parseInt(mklBlockTimeStr);
+        setNumThreads(getMklNumThreads());
+        setBlockTime(getMklBlockTime());
+        waitPolicy(getMklWaitPolicy());
+        if (getMklDisableFastMM()) {
+            disableFastMM();
+        }
+    }
 
+    public static int getMklNumThreads() {
         String mklNumThreadsStr = System.getProperty("bigdl.mklNumThreads", "1");
-        int mklNumThreads = Integer.parseInt(mklNumThreadsStr);
+        int num = Integer.parseInt(mklNumThreadsStr);
+        if (num <= 0) {
+            throw new UnsupportedOperationException("unknown bigdl.mklNumThreads " + num);
+        }
 
-        String mklDisableFastMMStr = System.getProperty("bigdl.mklDisableFastMM","true").toLowerCase();
-        boolean mklDisableFastMM = true;
+        return num;
+    }
+
+    public static int getMklBlockTime() {
+        String mklBlockTimeStr = System.getProperty("bigdl.mklBlockTime", "0");
+        int time = Integer.parseInt(mklBlockTimeStr);
+        if (time < 0) {
+            throw new UnsupportedOperationException("unknown bigdl.mklBlockTime " + time);
+        }
+
+        return time;
+    }
+
+    public static boolean getMklDisableFastMM() {
+        String mklDisableFastMMStr = System.getProperty("bigdl.mklDisableFastMM", "true").toLowerCase();
+        boolean mklDisableFastMM;
         if (mklDisableFastMMStr.equals("false")) {
             mklDisableFastMM = false;
+        } else if (mklDisableFastMMStr.equals("true")) {
+            mklDisableFastMM = true;
+        } else {
+            throw new UnsupportedOperationException("unknown bigdl.mklDisableFastMM " + mklDisableFastMMStr);
         }
 
-        String mklWaitPolicy = System.getProperty("bigdl.mklWaitPolicy", "passive").toLowerCase();
+        return mklDisableFastMM;
+    }
 
-        // set mkl environment variables
-        setNumThreads(mklNumThreads);
-        setBlockTime(mklBlockTime);
-        waitPolicyPassive();
+    public static int getMklWaitPolicy() {
+        String mklWaitPolicy = System.getProperty("bigdl.mklWaitPolicy", "passive").toLowerCase();
 
-        if (mklDisableFastMM) {
-            disableFastMM();
+        if (mklWaitPolicy.equalsIgnoreCase("passive")) {
+            return 3;
```

Comment:
define some static const variable to give the number with a name 

final static int MKL_WAIT_POLICY_PASSIVE = 3
final static int MKL_WAIT_POLICY_ACTIVE = 2

do not use magic number